src/ngraph/runtime/gpu/CMakeLists.txt

```diff
@@ -44,15 +44,103 @@ set(SRC
+        --compiler-options -fPIC;
+        -arch=sm_30;
+        -gencode=arch=compute_35,code=sm_35;
+        -gencode=arch=compute_50,code=sm_50;
+        -gencode=arch=compute_52,code=sm_52;
+        -gencode=arch=compute_60,code=sm_60;
+        -gencode=arch=compute_61,code=sm_61;
+        -gencode=arch=compute_61,code=compute_61)
+    if (CUDA9_FOUND)
+        set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};
+            -gencode=arch=compute_62,code=sm_62;
+            -gencode=arch=compute_70,code=sm_70;
+            -gencode=arch=compute_70,code=compute_70)
+    endif()
+
+    set (DO_CUDA_COMPILE FALSE)
+    if (CUDA9_FOUND)
+        if (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
+            # CUDA 9 supports up to gcc 6.x
+            if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+                set (DO_CUDA_COMPILE TRUE)
+            else()
+                message(STATUS "CUDA 9 only supports up to gcc 6.x")
+            endif()
+        elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+            # CUDA 9 supports up to clang 3.9
+            if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.0)
+                set (DO_CUDA_COMPILE TRUE)
+            else()
+                message(STATUS "CUDA 9 only supports up to clang 3.9")
+            endif()
+        endif()
+    else()
+        # CUDA 8 (minimum version of CUDA we support)
+        if (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
+            # Current release of CUDA 8 supports up to gcc 5.4
+            if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.5)
+                set (DO_CUDA_COMPILE TRUE)
+            else()
+                message(STATUS "CUDA 8 only supports up to gcc 5.4")
+            endif()
+        elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+            # CUDA 8 supports up to clang 3.8
+            if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.9)
+                set (DO_CUDA_COMPILE TRUE)
+            else()
+                message(STATUS "CUDA 8 only supports up to clang 3.8")
+            endif()
+        endif()
+    endif()
+
+    if (DO_CUDA_COMPILE)
+        if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+            # CUDA_PROPAGATE_HOST_FLAGS is true by default, so disable
+            # clang warnings that are known to flag CUDA code
+            set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};
+                --compiler-options -Wno-reserved-id-macro;
+                --compiler-options -Wno-undef;
+                --compiler-options -Wno-old-style-cast;
+                --compiler-options -Wno-deprecated;
+                --compiler-options -Wno-unused-macros;
+                --compiler-options -Wno-used-but-marked-unused)
+        endif()
+
+        message(STATUS "Doing cuda compile...")
+        set_source_files_properties( ${CUDA_SRC} PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)
+        cuda_include_directories(${CUDA_INC})
+        cuda_compile(CUDA_OBJ ${CUDA_SRC} STATIC)
+    else()
+        message(STATUS "Not doing cuda compile ")
```

Comment:
> Not doing cuda compile 
I think that these messages might be confusing considering most of our cuda kernels are built at runtime. If I saw cmake say that it was not doing cuda compilation, I might think there is not CUDA support via ngraph. So I propose to change all the message references to CUDA to something about NVCC. We could possibly have a message about kernels being compiled via NVRTC at runtime, e.g.
> "Not precompiling static CUDA kernels via NVCC, runtime compilation via NVRTC will be used." or something similar.

Reply:
maybe print info like "CUDA x and XXX is required for XXX, static CUDA kernels will not .."